spec/v2/providers/alerts/alerts.spec.ts

```diff
@@ -0,0 +1,235 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as config from '../../../../src/config';
+import { CloudEvent, CloudFunction } from '../../../../src/v2/core';
+import * as options from '../../../../src/v2/options';
+import * as alerts from '../../../../src/v2/providers/alerts';
+import {
+  BASIC_ENDPOINT,
+  BASIC_OPTIONS,
+  FULL_ENDPOINT,
+  FULL_OPTIONS,
+} from '../helpers';
+
+const ALERT_TYPE = 'new-alert-type';
+const APPID = '123456789';
+
+function getMockFunction(): CloudFunction<alerts.FirebaseAlertData<String>> {
+  const func = (raw: CloudEvent<unknown>) => 42;
+  func.run = (event: CloudEvent<alerts.FirebaseAlertData<String>>) => 42;
+  func.__trigger = 'silence the transpiler';
+  func.__endpoint = {};
+  return func;
+}
+
+describe('alerts', () => {
+  describe('onAlertPublished', () => {
+    it('should create the function without opts', () => {
+      const result = alerts.onAlertPublished(ALERT_TYPE, () => 42);
+
+      expect(result.__endpoint).to.deep.equal({
+        platform: 'gcfv2',
+        labels: {},
+        eventTrigger: {
+          eventType: alerts.eventType,
+          eventFilters: {
+            alertType: ALERT_TYPE,
+          },
+          retry: false,
+        },
+      });
+    });
+
+    it('should create the function with opts', () => {
+      const result = alerts.onAlertPublished(
+        {
+          ...BASIC_OPTIONS,
+          alertType: ALERT_TYPE,
+          appId: APPID,
+        },
+        () => 42
+      );
+
+      expect(result.__endpoint).to.deep.equal({
+        ...BASIC_ENDPOINT,
+        eventTrigger: {
+          eventType: alerts.eventType,
+          eventFilters: {
+            alertType: ALERT_TYPE,
+            appId: APPID,
+          },
+          retry: false,
+        },
+      });
+    });
+
+    it('should have a .run method', () => {
+      const func = alerts.onAlertPublished(ALERT_TYPE, (event) => event);
+
+      const res = func.run('input' as any);
+
+      expect(res).to.equal('input');
+    });
+  });
+
+  describe('defineEndpoint', () => {
```

Comment:
I kinda liked our way of testing high-level, external API (e.g. `onAlertPublished`) vs internal helper function like `defineEndpoint`. So we could just add a test case to check `__endpoint` prop is approriately set on each external API instead. WDYT?

Reply:
I think i now understand the purpose of defineEndpoint - it's meant to be used across all alerts API. This feels fine then. A minor suggestion for using `addEndpointAnnotation` to make it a little more obvious that calliing this fn has side-effect.

Reply:
I ended up changing the function to return the endpoint object (from your comment below) and re-named it to `getEndpointAnnotation`. How do you feel about that?